From d35a10dee269f7a780ec9560e9820f78d0968c0c Mon Sep 17 00:00:00 2001
From: Gergely Imreh <imrehg@gmail.com>
Date: Wed, 7 Jan 2009 13:45:20 +0800
Subject: [PATCH 2/2] Fix: regular expression matching of ffmpeg output

The regular expressions were borked, not sure if due to python  or ffmpeg
changes (since last version of TLF). Nevertheless, they were working for me
at the moment with ffmpeg Arch Linux compile, 2008-Jul-20 (unknown version).
---
 getFileInfo.py |   26 ++++++++++----------------
 main.py        |    4 +---
 2 files changed, 11 insertions(+), 19 deletions(-)

diff --git a/getFileInfo.py b/getFileInfo.py
index 6456c19..bbbd18c 100644
--- a/getFileInfo.py
+++ b/getFileInfo.py
@@ -5,33 +5,27 @@ def getInfo(file):
     try:
         cmd = ('ffmpeg -i "%s" -vframes 10 -vcodec mpeg4 -r 29.970030 -b 768k -ar 24000 -ab 128k -s 320x180 -f avi - > /dev/null' %(file))
         output = commands.getoutput(cmd)
-        print output
-        regexp = r'''(?isx)
-        .*?Duration:\s(?P<hours>\d{2}):(?P<minutes>\d{2}):(?P<seconds>\d{2}.\d),\s
-        '''
+
+        regexp = r'''(?isx).*?Duration:\s(?P<hours>\d{2}):(?P<minutes>\d{2}):(?P<seconds>\d{2}.\d)'''
         match = re.search(regexp, output)
         hours = int(match.group('hours'))
         minutes = int(match.group('minutes'))
         seconds = float(match.group('seconds'))
-        regexp = r'''(?isx)
-        .*?bitrate:\s(?P<bitrate>.*?)\s|$
-        '''
+
+        regexp = r'''(?isx).*?bitrate:\s(?P<bitrate>.*?)\s|$'''
         match = re.search(regexp, output)
         bitrate = match.group('bitrate')
-        regexp = r'''(?isx)
-        .*?Video:\s(?P<type>.*?),\s
-        '''
+
+        regexp = r'''(?isx).*?Video:\s(?P<type>.*?),\s'''
         match = re.search(regexp, output)
         type = match.group('type')
-        regexp = r'''(?isx)
-        .*?,\s(?P<width>\d*?)x(?P<height>\d*?),\s
-        '''
+
+        regexp = r'''(?isx).*?,\s(?P<width>\d*?)x(?P<height>\d*?)\s'''
         match = re.search(regexp, output)
         aspect = str(float(match.group('width'))/float(match.group('height')))[:6]
         size = match.group('width') + 'x' + match.group('height')
-        regexp = r'''(?isx)
-        .*?\s(?P<fps>\d{1,2}\.\d{1,2})\sfps
-        '''
+
+        regexp = r'''(?isx).*?\s(?P<fps>\d{1,2}\.\d{1,2})\stb'''
         match = re.search(regexp, output)
         fps = float(match.group('fps'))
         length = int((hours*3600 + minutes*60 + seconds)*1000)
diff --git a/main.py b/main.py
index 6bfed9c..b927788 100644
--- a/main.py
+++ b/main.py
@@ -2677,9 +2677,7 @@ class main(QDialog):
             total_frames += frames_count
         self.prog_dialog = QProgressDialog("Encoding ...", "Cancel Encoding",  total_frames, self, "progress", True)
         j = 0
-        regexp = r'''(?isx)
-        .*?frame=\s*?(?P<frames>\d*?)\sq=
-        '''
+        regexp = r'''(?isx).*?frame=\s*?(?P<frames>\d*?)\sfps='''
         cancelled = 0
         resultmessage = "The following files were successfully encoded:\n\n"
         for i in range(len(self.encodeSettings)):
-- 
1.6.1

